shapely/constructive.py

```diff
@@ -887,9 +887,21 @@ def simplify(geometry, tolerance, preserve_topology=True, **kwargs):
 def snap(geometry, reference, tolerance, **kwargs):
     """Snaps an input geometry to reference geometry's vertices.
 
-    The tolerance is used to control where snapping is performed.
+    Vertices of the first geometry are snapped to vertices of the second.
+    geometry, returning a new geometry; the input geometries are not modified.
     The result geometry is the input geometry with the vertices snapped.
     If no snapping occurs then the input geometry is returned unchanged.
+    The tolerance is used to control where snapping is performed.
+
+    Snapping one geometry to another can improve robustness for overlay
+    operations by eliminating nearly-coincident edges
+    (which cause problems during noding and intersection calculation).
```

Comment:
@brendan-ward this paragraph has been removed on the GEOS side (after your PR), so remove it here as well?

Reply:
Yes; was waiting for that to get updated before following up here.  I think lines 896 - 904 should be replaced with:

```
Where possible, this operation tries to avoid creating invalid geometries;
however, it does not guarantee that output geometries will be valid.  It is
the responsibility of the caller to check for and handle invalid geometries.

Because too much snapping can result in invalid
topology being created, heuristics are used to determine the number and
location of snapped vertices that are likely safe to snap. These heuristics
may omit some potential snaps that are otherwise within the tolerance.
```